extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaSecurityIdentityProvider.java

```diff
@@ -0,0 +1,9 @@
+package io.quarkus.amazon.lambda.http;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPEvent;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+public interface LambdaSecurityIdentityProvider {
```

Comment:
With the other changes I have suggested you should not need this, you can just use the existing `io.quarkus.security.identity.IdentityProvider`, but with a custom `io.quarkus.security.identity.request.AuthenticationRequest` that wraps the `APIGatewayV2HTTPEvent`